src/js/login.js

```diff
@@ -50,6 +42,71 @@ function showLoggedInBlock(show) {
         document.getElementById("password").value = "";
     }
 }
+
+function syncUserSettings() {
+    var listUserSettings = BASE_URL + "/aaa/listUserSettings.json?access_token=" + accessToken;
+    $.ajax({
+        url: listUserSettings,
+        dataType: "jsonp",
+        jsonpCallback: "p",
+        jsonp: "callback",
+        crossDomain: "true",
+        success: function(response) {
+            if (response.accepted) {
+                if (response.settings.theme != null) {
+                    var userThemeValue = response.settings.theme;
+                    if (defaultThemeValue !== userThemeValue) {
+                        defaultThemeValue = userThemeValue;
+                        if (defaultThemeValue === "dark") {
+                            chrome.storage.sync.set({
+                                "darktheme": true
+                            }, () => {});
+                        } else {
+                            chrome.storage.sync.set({
+                                "darktheme": false
+                            }, () => {});
+                        }
+                    }
+
+                }
+            }
+        }
+    });
+}
+
+function retrieveChatHistory() {
+    var serverHistoryEndpoint = BASE_URL + "/susi/memory.json?access_token=" + accessToken;
+    $.ajax({
+        url: serverHistoryEndpoint,
+        dataType: "jsonp",
+        jsonpCallback: "u",
+        jsonp: "callback",
+        crossDomain: "true",
+        success: function(response) {
+            var messages = [];
+            for (var i = response.cognitions.length - 1; i >= 0; i--) {
+                var queryAnswerPair = response.cognitions[i];
+                var queryTimes = new Date(Date.parse(queryAnswerPair.query_date));
+                var answerTimes = new Date(Date.parse(queryAnswerPair.answer_date));
+                //	console.table(response.cognitions[i]);
+                var queryInside = queryAnswerPair.query;
+                var answerInside = queryAnswerPair;
+                var msgObj = {
+                    query: queryInside,
+                    answer: answerInside,
+                    queryTime: queryTimes,
+                    answerTime: answerTimes
+                }
+                messages.push(msgObj);
+                localStorage.setItem("messages", JSON.stringify(messages));
+                console.table(messages);
```

Comment:
Why are you doing a `console.table` ?
Can you please elaborate the use it has here?

Reply:
i'll remove it.